### PR TITLE
fix scroll restoration issue on webKit browsers

### PR DESCRIPTION
**Pull request by martrapp**
## Changes

Scroll to a fragment did not work for webKit-based browsers. It seems that you need to explicitly set scrollRestoration to "auto" before the assignment to location.href (or location.hash) scrolls to the anchor. 

# Test
Tested manually as we do not have automated e2e tests on Safari.

# Doc
n./a. bug fix

### CODE REVIEW

File: .changeset/shaggy-socks-glow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+fix scroll restoration issue on webKit browsers
```
Comment (martrapp) [inline .changeset/shaggy-socks-glow.md:5]:
```suggestion
Fixed view transition issue on webKit browsers that prevented scrolling to #fragments 
```

File: .changeset/shaggy-socks-glow.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixed view transition issue on webKit browsers that prevented scrolling to #fragments
```
Comment (sarah11918) [inline .changeset/shaggy-socks-glow.md:5]:
```suggestion
Fixes a view transition issue on webKit browsers that prevented scrolling to #fragments 
```

**Review by sarah11918 — COMMENTED** [review]
Love the clarity of this message! Just updated the changeset to standardize on the verb tense we decided to use!

File: .changeset/shaggy-socks-glow.md (hunk 2 above)
Comment (martrapp) [reply .changeset/shaggy-socks-glow.md]: had a good trainer. I'm sure I'll get the times right at some point :)

**Review by MoustaphaDev — APPROVED** [review]
Makes sense to me!

**Pull request merged**